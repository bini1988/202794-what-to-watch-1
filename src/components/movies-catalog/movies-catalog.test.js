import React from "react";
import renderer from "react-test-renderer";
import MoviesCatalog from "./movies-catalog.jsx";
import genreGroupsMock from "../../mocks/movies-groups";

it(`MoviesCatalog correctly renders default markup`, () => {
  const tree = renderer
    .create(
        <MoviesCatalog
          moviesGenreGroups={genreGroupsMock}
          activeGenre="All Genres"
          onGenreChange={() => {}}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`MoviesCatalog correctly renders default markup with more button`, () => {
  const tree = renderer
    .create(
        <MoviesCatalog
          moviesGenreGroups={genreGroupsMock}
          activeGenre="All Genres"
          onGenreChange={() => {}}
          onMoviesMore={() => {}}/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
