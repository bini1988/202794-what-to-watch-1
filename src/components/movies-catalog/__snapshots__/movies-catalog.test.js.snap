// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesCatalog correctly renders default markup 1`] = `
<section
  className="catalog"
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item catalog__genres-item--active"
    >
      <a
        className="catalog__genres-link"
        onClick={[Function]}
      >
        All Genres
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      id="480d2236-6c7d-4fd3-a225-80c66710a71f"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height={175}
          muted={true}
          poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
          preload="none"
          width={280}
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/film/480d2236-6c7d-4fd3-a225-80c66710a71f"
        >
          Fantastic Beasts: The Crimes of Grindelwald
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      id="505f451a-0b69-47ed-96c6-66d5525d5a5e"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          height={175}
          muted={true}
          poster="img/bohemian-rhapsody.jpg"
          preload="none"
          width={280}
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/film/505f451a-0b69-47ed-96c6-66d5525d5a5e"
        >
          Bohemian Rhapsody
        </a>
      </h3>
    </article>
  </div>
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      type="button"
    >
      Show more
    </button>
  </div>
</section>
`;
